ott/core/nn_potentials/icnn.py

```diff
@@ -0,0 +1,135 @@
+# coding=utf-8
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Implementation of Amos+(2017) input convex neural networks (ICNN)."""
+
+import jax
+import jax.numpy as jnp
+from flax import linen as nn
```

Comment:
It seems this should be added to the dependencies of the project?

Reply:
Shall I add flax to the `requirements.txt`?

Reply:
yes I think we will need it.